Optimized the printing of versions used Hashset

vishalsinghji opened the pull request.

acheroncrypto reviewed (commented):
Thanks but what are we optimizing here actually?

vishalsinghji commented:
> Thanks but what are we optimizing here actually?

1. Used a HashSet to efficiently check if a version is installed.
2. Extracted the common print logic into a helper function (print_version_info).
3. Simplified the loop and conditionals for printing available versions.
4. Removed unnecessary cloning and modified the iteration to work directly with references.
5. Consolidated and cleaned up the code for better readability.

vishalsinghji commented:
@acheroncrypto  Can you Review this PR ?
